### PR TITLE
Publisher Channels V3 API

### LINKED ISSUE #2142: Create a v3 publishers channels API
We should introduce a property that defines whether or not a publisher is KYC'd, verified or unverified

```
["yachtcaptain23.github.io",true,false,"7d9c7b8a-12fd-4235-9238-f14f8ef5f5de",{}]
```
```
["yachtcaptain23.github.io","verified",false,"7d9c7b8a-12fd-4235-9238-f14f8ef5f5de",{}]
```

### PR DESCRIPTION
**Pull request by corymcdonald**
## Publisher Channels V3 API
Closes #2142 

#### Features

- Adds support for non-KYC'd publishers to have their tips held as "pending" on the browser
- Generates statistical totals based on the JSON and not what we have in our database (Reflective of the actual system) 
- Reduces the amount of queries executed / minor performance gains

| Version | Time (seconds)| Channels Generated|
| -------- | ---- | ---------------------- |
|  V1 | 1524.278572 | 251,094 |
| V2 | 1244.177728  | 83,917 |
| V3 | 1617.463192 | 251,094 | 

<i>* Generated using Ruby Benchmark using Production data</i>

Here's the results from the latest `/api/v3/channels/totals` endpoint
```JSON

{
  "site": {
    "verified": 12935,
    "connected": 11830,
    "total": 24765
  },
  "all_channels": {
    "verified": 83927,
    "connected": 97974,
    "total": 181901
  },
  "twitch": {
    "verified": 5586,
    "connected": 4964,
    "total": 10550
  },
  "github": {
    "connected": 481,
    "verified": 488,
    "total": 969
  },
  "youtube": {
    "verified": 56411,
    "connected": 71855,
    "total": 128266
  },
  "reddit": {
    "verified": 1303,
    "connected": 1164,
    "total": 2467
  },
  "vimeo": {
    "connected": 160,
    "verified": 236,
    "total": 396
  },
  "twitter": {
    "verified": 6968,
    "connected": 7520,
    "total": 14488
  }
}
```

#### How To Test

Hit the following URLs

https://localhost:3000/api/v3/public/channels/
https://localhost:3000/api/v3/public/channels/totals

### CODE REVIEW

File: test/fixtures/uphold_connection_for_channels.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+default: &default
```
Comment (yachtcaptain23) [inline test/fixtures/uphold_connection_for_channels.yml]: I think we should be renaming this, but it somehow works? I thought there would be a naming conflict where we use `default` for a publisher (even though the proper name should be `default_publisher` and we'd name this `default_uphold_connection_for_channel`

File: app/models/json_builders/channels_json_builder_v3.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      wallet_address_id,
+      site_banner_details(verified_channel),
+    ])
+  end
+
+  def status(verified_channel, address)
+    connection = verified_channel.publisher&.uphold_connection
+
+    if connection&.is_member && address.present?
+      "verified"
```
Comment (yachtcaptain23) [inline app/models/json_builders/channels_json_builder_v3.rb]: Maybe constants?

Comment (corymcdonald) [reply app/models/json_builders/channels_json_builder_v3.rb]: Added 
https://github.com/brave-intl/publishers/pull/2150/commits/bbe5ca8e155b51399bc56580afe3ee12375fc1d7

File: test/fixtures/uphold_connection_for_channels.yml (hunk 1 above)
Comment (corymcdonald) [reply test/fixtures/uphold_connection_for_channels.yml]: https://github.com/brave-intl/publishers/pull/2150/commits/bbe5ca8e155b51399bc56580afe3ee12375fc1d7

**Pull request merged**